server/chat/backend/agent/tools/cloud_tools.py

```
@@ -1305,6 +1305,32 @@ def cloud_exec_wrapper(provider: str, command: str, output_file: Optional[str] =
+    try:
+        from .slack_tool import (
+            list_slack_channels,
+            get_channel_history,
+            get_thread_replies,
+            is_slack_connected,
+        )
+        if user_id and is_slack_connected(user_id):
+            tool_functions.append((list_slack_channels, "list_slack_channels"))
+            tool_functions.append((get_channel_history, "get_channel_history"))
+            tool_functions.append((get_thread_replies, "get_thread_replies"))
+            logging.info(f"Added Slack tools for user {user_id}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)